File: descending-data/InsertionSort/InsertionSort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void insertionSort(int arr[], int n) {
    int i, key, j;
    for (i = 1; i < n; i++) {
        key = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

int main() {
    FILE *file = fopen("descending_data.txt", "r");
    if (file == NULL) {
        printf("Error opening file!\n");
        return 1;
    }

    int arr[25000], n = 0;

    // Read integers from the file
    while (fscanf(file, "%d", &arr[n]) != EOF) {
        n++;
    }
    fclose(file);

    // Measure sorting time
    clock_t start, end;
    start = clock();
    insertionSort(arr, n);
    end = clock();

    // Calculate time taken
    double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;

    

    printf("Execution Time: %f seconds\n", time_taken);
    return 0;
}

File: descending-data/InsertionSort/InsertionSort.py
import time
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
file_name = "descending_data.txt"
try:
    with open(file_name, "r") as file:
        data = file.read()
        arr = list(map(int, data.split()))
except FileNotFoundError:
    print("Error: descending_data.txt file not found!")
    exit()
start_time = time.time()
insertion_sort(arr)
end_time = time.time()
execution_time = end_time - start_time
print(f"Insertion Sort Execution Time: {execution_time * 1000:.4f} milliseconds")